src/rpc/core_rpc_server.cpp

```diff
@@ -3359,51 +3359,52 @@ namespace cryptonote
         return false;
 
       std::vector<lns::mapping_record> records = db.get_mappings(request.types, request.name);
       res.entries.reserve(records.size());
       for (auto const &record : records)
       {
         res.entries.emplace_back();
-        COMMAND_RPC_GET_LNS_NAMES_TO_OWNERS::response_entry &entry = res.entries.back();
-        entry.entry_index         = request_index;
-        entry.type                = static_cast<uint16_t>(record.type);
-        entry.owner               = epee::string_tools::pod_to_hex(record.owner);
-        entry.value               = extract_lns_mapping_value(record);
-        entry.register_height     = record.register_height;
-        entry.txid                = epee::string_tools::pod_to_hex(record.txid);
-        entry.prev_txid           = epee::string_tools::pod_to_hex(record.prev_txid);
+        COMMAND_RPC_LNS_NAMES_TO_OWNERS::response_entry &entry = res.entries.back();
+        entry.entry_index                                      = request_index;
+        entry.type                                             = static_cast<uint16_t>(record.type);
+        entry.owner                                            = epee::string_tools::pod_to_hex(record.owner);
+        entry.backup_owner                                     = epee::string_tools::pod_to_hex(record.backup_owner);
+        entry.value                                            = extract_lns_mapping_value(record);
+        entry.register_height                                  = record.register_height;
+        entry.txid                                             = epee::string_tools::pod_to_hex(record.txid);
+        entry.prev_txid                                        = epee::string_tools::pod_to_hex(record.prev_txid);
       }
     }
 
     res.status = CORE_RPC_STATUS_OK;
     return true;
   }
   //------------------------------------------------------------------------------------------------------------------------------
-  bool core_rpc_server::on_get_lns_owners_to_names(const COMMAND_RPC_GET_LNS_OWNERS_TO_NAMES::request &req, COMMAND_RPC_GET_LNS_OWNERS_TO_NAMES::response &res, epee::json_rpc::error &error_resp, const connection_context *ctx)
+  bool core_rpc_server::on_lns_owners_to_names(const COMMAND_RPC_LNS_OWNERS_TO_NAMES::request &req, COMMAND_RPC_LNS_OWNERS_TO_NAMES::response &res, epee::json_rpc::error &error_resp, const connection_context *ctx)
   {
-    if (exceeds_quantity_limit(ctx, error_resp, m_restricted, req.entries.size(), COMMAND_RPC_GET_LNS_OWNERS_TO_NAMES::MAX_REQUEST_ENTRIES))
+    if (exceeds_quantity_limit(ctx, error_resp, m_restricted, req.entries.size(), COMMAND_RPC_LNS_OWNERS_TO_NAMES::MAX_REQUEST_ENTRIES))
       return false;
 
-    std::map<crypto::ed25519_public_key, size_t> key_to_request_index;
-    std::vector<crypto::ed25519_public_key> keys;
+    std::map<crypto::generic_public_key, size_t> key_to_request_index;
```

Comment:
Should be an unordered_map

Reply:
unordered_map on our pubkey types is actually really fast because we don't actually have to hash anything, we just use the value itself as a hash.  But you're right, for a small container it's not going to matter anyway.